integration/experiment/gpu_frequency_sweep/gpu_frequency_sweep.py

```diff
@@ -0,0 +1,125 @@
+from experiment import machine
+from experiment.frequency_sweep import frequency_sweep
+
+
+def setup_run_args(parser):
+    common_args.setup_run_args(parser)
+    common_args.add_run_max_turbo(parser)
+    common_args.add_min_frequency(parser)
+    common_args.add_max_frequency(parser)
+    common_args.add_step_frequency(parser)
+    parser.add_argument('--max-gpu-frequency', dest='max_gpu_frequency',
+                        action='store', type=float, default=None,
+                        help='top gpu frequency setting for the sweep')
+    parser.add_argument('--min-gpu-frequency', dest='min_gpu_frequency',
+                        action='store', type=float, default=None,
+                        help='bottom gpu frequency setting for the sweep')
+    parser.add_argument('--step-gpu-frequency', dest='step_gpu_frequency',
+                        action='store', type=float, default=None,
+                        help='increment in hertz between gpu frequency settings for the sweep')
+
+
+def setup_gpu_frequency_bounds(mach, min_gpu_freq, max_gpu_freq,
+                                  step_gpu_freq):
+
+    #TODO: Don't hardcode this. What.
+    sys_min = 1.2e9
+    sys_max = 2.7e9
+    sys_step = mach.frequency_step()
+    if min_gpu_freq is None:
+        min_gpu_freq = sys_min
+    if max_gpu_freq is None:
+        max_gpu_freq = sys_max
+    if step_gpu_freq is None:
+        step_gpu_freq = sys_step
+    if step_gpu_freq < sys_step or step_gpu_freq % sys_step != 0:
+        sys.stderr.write('<geopm> Warning: gpu frequency step size may be incompatible with p-states.\n')
+    if (max_gpu_freq - min_gpu_freq) % step_gpu_freq != 0:
+        sys.stderr.write('<geopm> Warning: gpu frequency range not evenly divisible by step size.\n')
+    if min_gpu_freq < sys_min or max_gpu_freq > sys_max:
+        raise RuntimeError('GPU frequency bounds are out of range for this system')
+    if min_gpu_freq > max_gpu_freq:
+        raise RuntimeError('GPU frequency min is greater than max')
+
+    num_step = 1 + int((max_gpu_freq - min_gpu_freq) // step_gpu_freq)
+    gpu_freqs = [step_gpu_freq * ss + min_gpu_freq for ss in range(num_step)]
+    gpu_freqs = sorted(gpu_freqs, reverse=True)
+    return gpu_freqs
+
+
+def report_signals():
+    return ["CPU_CYCLES_THREAD@package", "CPU_CYCLES_REFERENCE@package",
+            "TIME@package", "CPU_ENERGY@package", "GPU_CORE_FREQUENCY_STATUS@board"]
+
+
+def trace_signals():
+    return [ "GPU_CORE_FREQUENCY_STATUS@gpu" ]
+
+def launch_configs(app_conf, core_freq_range, gpu_freq_range):
+    agent = 'frequency_map'
+    targets = []
+    for freq in core_freq_range:
+        for gpu_freq in gpu_freq_range:
+            name = '{:.1e}c_{:.1e}u'.format(freq, gpu_freq)
+            options = {'FREQ_DEFAULT': freq,
+                       'FREQ_GPU_DEFAULT': gpu_freq}
+            agent_conf = geopmpy.agent.AgentConf('{}_agent_{}c_{}u.config'.format(agent, freq, gpu_freq), agent, options)
```

Comment:
I think `{}c` in the frequency-sweep script was CPU core frequency and `{}u` was CPU uncore frequency. Maybe in this script `c{}` could be CPU, but if that's the case, then `u{}` should probably be `g{}` for GPU.